plugins/scaffolder/src/components/ScaffolderFilter/ScaffolderFilter.test.tsx

```diff
@@ -0,0 +1,264 @@
+  identityApiRef,
+  storageApiRef,
+} from '@backstage/core';
+import { catalogApiRef } from '@backstage/plugin-catalog-react';
+import { MockStorageApi, wrapInTestApp } from '@backstage/test-utils';
+import { fireEvent, render, waitFor } from '@testing-library/react';
+import React from 'react';
+import { EntityFilterGroupsProvider } from '../../filter';
+import { ButtonGroup, ScaffolderFilter } from './ScaffolderFilter';
+
+describe('Catalog Filter', () => {
+  const catalogApi: Partial<CatalogApi> = {
+    getEntities: () =>
+      Promise.resolve({
+        items: [
+          {
+            apiVersion: 'backstage.io/v1alpha1',
+            kind: 'Component',
+            metadata: {
+              name: 'Entity1',
+            },
+            spec: {
+              owner: 'tools@example.com',
+              type: 'service',
+            },
+          },
+          {
+            apiVersion: 'backstage.io/v1alpha1',
+            kind: 'Component',
+            metadata: {
+              name: 'Entity2',
+            },
+            spec: {
+              owner: 'not-tools@example.com',
+              type: 'service',
+            },
+          },
+        ] as Entity[],
+      }),
+  };
+
+  const identityApi: Partial<IdentityApi> = {
+    getUserId: () => 'tools@example.com',
+  };
+
+  const renderWrapped = (children: React.ReactNode) =>
+    render(
+      wrapInTestApp(
+        <ApiProvider
+          apis={ApiRegistry.from([
+            [catalogApiRef, catalogApi],
+            [identityApiRef, identityApi],
+            [storageApiRef, MockStorageApi.create()],
+          ])}
+        >
+          <EntityFilterGroupsProvider>{children}</EntityFilterGroupsProvider>,
+        </ApiProvider>,
+      ),
+    );
+
+  it('should render the different groups', async () => {
+    const mockGroups: ButtonGroup[] = [
+      { name: 'Test Group 1', items: [] },
+      { name: 'Test Group 2', items: [] },
+    ];
+    const { findByText } = renderWrapped(
+      <ScaffolderFilter buttonGroups={mockGroups} initiallySelected="" />,
+    );
+    for (const group of mockGroups) {
+      expect(await findByText(group.name)).toBeInTheDocument();
+    }
+  });
+
+  it('should render the different items and their names', async () => {
+    const mockGroups: ButtonGroup[] = [
+      {
+        name: 'Test Group 1',
+        items: [
+          {
+            id: 'all',
+            label: 'First Label',
+            filterFn: () => true,
+          },
+          {
+            id: 'starred',
+            label: 'Second Label',
+            filterFn: () => false,
+          },
+        ],
+      },
+    ];
+
+    const { findByText } = renderWrapped(
+      <ScaffolderFilter buttonGroups={mockGroups} initiallySelected="all" />,
+    );
+
+    for (const item of mockGroups[0].items) {
+      expect(await findByText(item.label)).toBeInTheDocument();
+    }
+  });
+
+  it('selects the first item if no desired initial one is set', async () => {
```

Comment:
I'd add two more describes one for groups one for items to make it a bit clearer what piece of functionality it is that we're testing.